library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/storage/SessionReplayRecordWriter.kt

```diff
@@ -7,16 +7,21 @@
 package com.datadog.android.sessionreplay.internal.storage
 
 import com.datadog.android.sessionreplay.SessionReplayFeature
+import com.datadog.android.sessionreplay.internal.RecordCallback
 import com.datadog.android.sessionreplay.internal.RecordWriter
 import com.datadog.android.sessionreplay.internal.processor.EnrichedRecord
 import com.datadog.android.v2.api.SdkCore
 
-internal class SessionReplayRecordWriter(private val sdkCore: SdkCore) : RecordWriter {
+internal class SessionReplayRecordWriter(
+    private val sdkCore: SdkCore,
+    private val recordCallback: RecordCallback
+) : RecordWriter {
     private var lastRumContextId: String = ""
     override fun write(record: EnrichedRecord) {
         val forceNewBatch = resolveForceNewBatch(record)
         sdkCore.getFeature(SessionReplayFeature.SESSION_REPLAY_FEATURE_NAME)
             ?.withWriteContext(forceNewBatch) { _, eventBatchWriter ->
+                updateViewSent(record)
```

Comment:
probably it better to place this call below in the following way:

```
if (eventBatchWriter.write(serializedRecord, null)) {
  updateViewSent(record)
}
```
that way it will be invoked only if record was really written. Otherwise, I guess, you may have some discrepancy if write operation failed.